x-pack/plugins/infra/public/pages/metrics/hosts/components/cloud_provider_icon_with_title.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiText } from '@elastic/eui';
+import React from 'react';
+
+const cloudIcons: Record<string, string> = {
+  gcp: 'logoGCP',
+  aws: 'logoAWS',
+  azure: 'logoAzure',
```

Comment:
@formgeist really nice idea, thanks! I added the fallback [here](https://github.com/elastic/kibana/pull/147516/commits/f8ee9e9a237156a6b96f64ab20f6e780c8d06b10) and it will look like this: 
<img width="759" alt="image" src="https://user-images.githubusercontent.com/14139027/207627541-de1919bd-dc97-499b-9d67-d0c385b02b32.png">

Reply:
@jennypavlova Looks great! Thanks for updating it 👍  it looks like the text size of the name column is larger than the rest? Can we decrease it? I support the icons would also need to be downsized to fit. Sorry for the nit-picking 🎨

Reply:
@formgeist Thank you for spotting that! Now the hostname has the size defined by its container and the icon is downsized to `m`. 
![image](https://user-images.githubusercontent.com/14139027/207643354-6558429b-edb3-436f-b458-d8320c9aaad9.png)

Reply:
@jennypavlova Awesome - I added a new issue for adding a tooltip to the icon https://github.com/elastic/kibana/issues/147568 - in case we don't have the icon or users don't recognize the icon, it'd be great to show the cloud.provider value.